File: Interpreter C#/Interpreter/closeRFile.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Interpreter
{
    namespace Model
    {
       
        class closeRFile : IStmt
        {
            Exp exp_file_id;

            public closeRFile(Exp e)
            {
                this.exp_file_id = e;
            }

            PrgState IStmt.execute(PrgState state)
            {
                //IStmt stmt = state.getExeStack().pop();
                int val = this.exp_file_id.eval(state.getSymTable());
                Tuple<String, StreamReader> t = state.getFileTable().lookup(val);
                if (t == null)
                {
                    return null;
                }
                else
                {
                    t.Item2.Close();
                    state.getFileTable().remove(val);
                }
                return state;
            }

            public override string ToString()
            {
                return "close file :" + this.exp_file_id.ToString() + ") \n";
            }
        }
    }
}

File: MAP/Interpretor2/src/Model/closeRFile.java
package Model;

import java.io.BufferedReader;

public class closeRFile implements IStmt {
	
	public Exp exp_file_id;
	
	public closeRFile(Exp e)
	{
		this.exp_file_id=e;
	}
	public String toString()
	{
		return "Close file ("+this.exp_file_id.toString()+" )";
	}

	public Exp getExp()
	{
		return this.exp_file_id;
	}
	
	@Override
	public PrgState execute(PrgState state) {
		try {
			Integer val=this.exp_file_id.eval(state.symTable,state.heap);
			Tuple<String,BufferedReader> t=state.fileTable.lookup(val);
			if(t==null)
			{
				return null;
			}
			else
			{
				t.descriptor.close();
				state.fileTable.remove(val);
			}
		} catch (Exception e) {
			System.out.println(e);
			e.printStackTrace();
		}
		return state;
		
	}

}
